bookkeeper-server/src/main/java/org/apache/bookkeeper/util/DiskChecker.java

```diff
@@ -155,6 +156,29 @@ float checkDiskFull(File dir) throws DiskOutOfSpaceException, DiskWarnThresholdE
     }
 
     /**
+     * calculates and returns the disk usage factor in the provided list of dirs
+     * 
+     * @param dirs
+     *            list of directories
+     * @return disk usage factor in the provided list of dirs
+     */
+    public float getTotalDiskUsage(List<File> dirs) {
+        if (dirs == null || dirs.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "list argument of getTotalDiskUsage is not supposed to be null or empty");
+        }
+        long totalUsableSpace = 0;
+        long totalSpace = 0;
+        for (File dir : dirs) {
+            totalUsableSpace += dir.getUsableSpace();
+            totalSpace += dir.getTotalSpace();
```

Comment:
very minor: These will only work if each of the directories are on separate partitions, which is what a user should be doing anyways.  It might be worth using the Files API with FileStore to dedupe these, or at least output some sort of a warning somewhere if they are not the separate partitions.